packages/backend/src/api/controllers/events/EventsController.ts

```diff
@@ -0,0 +1,110 @@
+import {
+  ApiEvents,
+  EventChart,
+  EventChartPoint,
+  ProjectId,
+  UnixTime,
+} from '@l2beat/types'
+
+import { EventUpdater } from '../../../core/events/EventUpdater'
+import { ProjectInfo } from '../../../model'
+import { EventRepository } from '../../../peripherals/database/EventRepository'
+import {
+  getHourlyMinTimestamp,
+  getSixHourlyMinTimestamp,
+} from '../report/charts'
+import { renderShowcasePage } from './ShowcasePage'
+
+export class EventsController {
+  constructor(
+    private eventsRepository: EventRepository,
+    private eventUpdater: EventUpdater,
+    private projects: ProjectInfo[],
+  ) {}
+
+  async getEvents(): Promise<ApiEvents> {
+    const main: ApiEvents = {
+      projects: {},
+    }
+
+    const timestamp = this.eventUpdater.getLastProcessed()
+
+    const config: {
+      granularity: 'hourly' | 'sixHourly' | 'daily'
+      timestamp: UnixTime | undefined
+    }[] = [
+      { granularity: 'hourly', timestamp: getHourlyMinTimestamp(timestamp) },
+      {
+        granularity: 'sixHourly',
+        timestamp: getSixHourlyMinTimestamp(timestamp),
+      },
+      { granularity: 'daily', timestamp: undefined },
+    ]
+
+    await Promise.all(
+      this.projects.map(async ({ projectId, events }) => {
+        const eventNames = events.map((e) => e.name)
+
+        const eventCharts = await Promise.all(
+          config.map((c) =>
+            this.getEventChart(
+              projectId,
+              c.granularity,
+              c.timestamp,
+              eventNames,
+            ),
+          ),
+        )
+
+        main.projects[projectId.toString()] = {
+          hourly: eventCharts[0],
+          sixHourly: eventCharts[1],
+          daily: eventCharts[2],
+        }
+      }),
+    )
+    return main
+  }
+
+  async getEventChart(
+    projectId: ProjectId,
+    granularity: 'hourly' | 'sixHourly' | 'daily',
+    minTimestamp: UnixTime | undefined,
+    eventNames: string[],
+  ): Promise<EventChart> {
+    const records = await this.eventsRepository.getByProject(
+      projectId,
+      granularity,
+      minTimestamp,
+    )
+
+    //Record<timestamp, Record<eventName, count>>
```

Comment:
I know that a good code does not need a comment, but here it is very useful in my opinion. Other way around there would be needed more types and logic which is not needed here if it can be solved with a really barebone type and a comment. But I am open to discussing it.